### PR TITLE
Run all integration-cli daemons in separate containers

### PR DESCRIPTION
**Pull request by tonistiigi**
This PR changes the integration-cli tests so that every daemon runs in their own container. This is a prerequisite for writing actual networking tests and testing new swarm features like virtual IPs and routing mesh.

I tried to avoid building new dind image every time because we want to test dynamic binaries, meaning we would have to maintain a full docker runtime context base image, include that as a frozen image and update and change that every time any dependency changes in master. Using new image would also be slower and lots of tests already share files between test and the daemon.

I also found some bugs in the tests that only worked because of some side effect of the previous framework. There are places that can be improved now with much fewer lines but I tried to avoid making any meaningful changes to existing tests. Lots of networking tests are safer now and don't share interfaces with the host but there are still a bunch that used the main daemon or even mix between them. All these should be unified later, for example, I found we have tests atm that can't be run 2 times in a row because they leave incomplete state to the main daemon.

The change should normally not add very much overhead but our CI runs with vfs graphdriver that is very slow on starting a container. I built a static binary nsenter image https://github.com/tonistiigi/nsenter that could be used to avoid the slow data copy in vfs if we include it as a frozen image in Dockerfile. I estimate this would save us 1-2sec per daemon in CI. Let me know if you think its good idea to switch to that image. 

New Virtual IP and Routing mesh integration test is also added. Routing mesh test needed some sleep calls to pass consistently.

[wip] This PR requires a libnetwork fix to not use hardcoded path for storing netns fds. I included a quick patch in here for now so we can see the tests working.

@aboch @mrjana @mavenugo

### CODE REVIEW

**Comment by thaJeztah** [comment]
> [wip] This PR requires a libnetwork fix to not use hardcoded path for storing netns fds. I included a quick patch in here for now so we can see the tests working.

@tonistiigi is that https://github.com/docker/docker/issues/18508#issuecomment-232669207 ?

> The change should normally not add very much overhead but our CI runs with vfs graphdriver that is very slow on starting a container.

Didn't we make a change to have `/var/lib/docker` run in a volume, so that other graph drivers could be used?

**Comment by tonistiigi** [comment]
@thaJeztah 
It is similar but in this specific case the problem is that the exec-root parameter is not used instead of `/var/run/docker/`.

I'm not sure what change you have in mind with the volume. `/var/lib/docker` is already a volume in our Dockerfile. The daemons in this PR work fine with overlay/aufs locally. Getting them to work with drivers backed by loop devices would be much harder/slower and we don't support them atm as well. Our CI is set to use vfs, we could switch it to aufs/overlay if we wanted.

**Comment by thaJeztah** [comment]
> we could switch it to aufs/overlay if we wanted.

ah, yes, that's what I meant; I thought that we already changed that

**Comment by tonistiigi** [comment]
@mavenugo Should I create a separate issue for this sleep https://github.com/docker/docker/pull/24613/files#diff-248890d27a734b6320cc5296c4695f1dR109 ? I've also seen this test fail once with tcp dial timeout so probably the same issue.

Experimental suite has some weird issues. It seems to lose network connectivity while running the macvlan tests. This is strange because these tests use the main daemon and are unchanged. I wonder if they passed before because of some combination with another test. In my debug `TestDockerNetworkMacVlanExistingParent` seemed to be the first test that caused this.

**Comment by mavenugo** [comment]
@tonistiigi 

> Should I create a separate issue for this sleep

pls go ahead.

we should remove macvlan tests from experimental and make it part of default builds.
ping @nerdalert to take a look at these failures.

**Comment by icecrime** [comment]
Removing from milestone as 1) Tonis is in vacations 2) I don't think this is strictly required for the release.

File: integration-cli/check_test.go [hunk 1]
```diff
@@ -175,6 +175,7 @@ type DockerDaemonSuite struct {
 
 func (s *DockerDaemonSuite) SetUpTest(c *check.C) {
 	testRequires(c, DaemonIsLinux)
```
Comment (jhowardmsft) [inline integration-cli/check_test.go]: These two lines can be combined

**Comment by vdemeester** [comment]
@tonistiigi what is the status here ? 👼

**Comment by tonistiigi** [comment]
@vdemeester  Experimental `MacVlan` tests still have the problem I described above. Everything else was passing.

**Pull request closed without merging**